src/zenml/pipelines/base_pipeline.py

```diff
@@ -260,7 +269,40 @@ def requirements(self) -> Set[str]:
                     f"'{integration_name}'."
                 ) from e
 
-        if self.requirements_file and fileio.exists(self.requirements_file):
+        if isinstance(self._requirements, str) and fileio.exists(
+            self._requirements
+        ):
+            with fileio.open(self._requirements, "r") as f:
+                requirements.update(
+                    {
+                        requirement.strip()
+                        for requirement in f.read().split("\n")
+                    }
+                )
+        # Add this logic back in (described in #ENG-882)
+        #
+        # elif isinstance(self._requirements, str) and self._requirements:
+        #     root = str(Repository().root)
+        #     if root:
+        #         assumed_requirements_file = os.path.join(
+        #             root, "requirements.txt"
+        #         )
+        #         if fileio.exists(assumed_requirements_file):
+        #             with fileio.open(assumed_requirements_file, "r") as f:
+        #                 requirements.update(
+        #                     {
+        #                         requirement.strip()
+        #                         for requirement in f.read().split("\n")
+        #                     }
+        #                 )
+        #                 logger.info(
+        #                     "Using requirements file: `%s`",
+        #                     assumed_requirements_file,
+        #                 )
+        elif isinstance(self._requirements, List):
+            requirements.update(self._requirements)
+        elif self.requirements_file and fileio.exists(self.requirements_file):
```

Comment:
I have one more request/question: With the current implementation if you set `requirements` then `requirements_`file` is simply ignored. Is this intended, and if yes should we log a warning message here that it's ignored? I can see someone just seeing the argument names and specifying both which would result in "unexpected" behaviour?

Reply:
Good point.